vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,6 +207,315 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
       updateIntegrationStatus(vc);
    }
 
+   public VsphereClient connectVsphere(SDDCSoftwareConfig vc) throws Exception {
+      return VsphereClient.connect(String.format(VCConstants.SDKURL, vc.getServerURL()),
+            vc.getUserName(), vc.getPassword(), !vc.isVerifyCert());
+   }
+
+   public HashMap<String, ServerMapping> getVaildServerMapping(SDDCSoftwareConfig vc) {
+
+      HashMap<String, ServerMapping> mobIdDictionary = new HashMap<String, ServerMapping>();
+      ServerMapping[] mappings = null;
+      try {
+         restClient.setServiceKey(serviceKeyConfig.getServiceKey());
+         mappings = restClient.getServerMappingsByVC(vc.getId()).getBody();
+      } catch (HttpClientErrorException clientError) {
+         if (clientError.getRawStatusCode() != HttpStatus.NOT_FOUND.value()) {
+            return null;
+         }
+      }
+
+      for (ServerMapping mapping : mappings) {
+         if (mapping.getAsset() != null) {
+            mobIdDictionary.put(mapping.getVcMobID(), mapping);
+         }
+      }
+      return mobIdDictionary;
+
+   }
+   public void queryHostMetrics(SDDCSoftwareConfig vc) {
+
+      HashMap<String, ServerMapping> serverMappingMap = getVaildServerMapping(vc);
+      if (serverMappingMap == null) {
+         logger.info("serverMapping is invaild");
+         return;
+      }
+      try (VsphereClient vsphereClient = connectVsphere(vc);) {
+
+         Collection<HostSystem> hosts = vsphereClient.getAllHost();
+         if (hosts == null || hosts.isEmpty()) {
+            logger.error("vsphere: " + vsphereClient.getVCUUID() + " get hosts is null");
+            return;
+         }
+         for (HostSystem host : hosts) {
+
+            String mobId = host._getRef().getValue();
+            if (serverMappingMap.containsKey(mobId)) {
+               ServerMapping serverMapping = serverMappingMap.get(mobId);
+               String assetId = serverMapping.getAsset();
+
+               Asset hostMappingAsset = restClient.getAssetByID(assetId).getBody();
+               if (hostMappingAsset == null) {
+                  logger.error("serverMapping: " + serverMapping.getId() + " get asset: " + assetId + " is null");
```

Comment:
logger.error("serverMapping:{} get asset: {} is null", serverMapping.getId(), assetId);